### PR TITLE
Include Secrets Manager for Lambda

### PR DESCRIPTION
**Pull request by tmaior**

### CODE REVIEW

**Review by zach-withcoherence — CHANGES_REQUESTED** [review]
thanks for this. so you don't actually need to make any new secrets for this, just need to add them to the environment on each deploy like this for ecs for example https://github.com/coherenceplatform/cnc/blob/main/src/cnc/flavors/aws/ecs/1/deploy/ecs_web_task.json.j2#L44C1-L49C29

cnc does not create cloud secrets except for the managed ones which are already there for DBs, can just reference secret IDs in the environments.yml and those will be added as well

**Comment by tmaior** [comment]
Perfect. I'll adjust it today.

**Comment by tmaior** [comment]
One thing I noticed is that in the example, it already uses the Secret Manager arn, to get the values ​​through Valuefrom, with Valuefrom it can get the values.

However, valuefrom is only for the ECS Task Definition.

In the lambda, we would have to pass the values ​​in the variables.

In this case, would we have to use the SecretsManager datasource?

Like this one in the example in the print. (Just test it so we can align)
image

**Comment by zach-withcoherence** [comment]
We have 3 options here I guess:
- we fetch the value during the deploy and pass the secret value as a plain environment variable. This is easy in the concept script since item.value knows how to fetch the secret value already, we just need to include secret items in the loop in the script
- we provide the secret ARN as the environment variable value, and assume the user will fetch the secret values from the ARN at runtime
- something fancier would be to inject some layer we manage that fetches the values and sets them as environment variables but might be too magical

@adamaziz15 i think option 1 has security issues but is ultimately what people want. @tmaior what do you think here

**Comment by tmaior** [comment]
@zach-withcoherence 

1. There is indeed a security concern.

2. The main issue here is that the responsibility of providing the secret's ARN falls on the user. In terms of implementation, this is much simpler. My opinion: since we are not creating the Secret for Serverless, I believe this is a good option. Considering it's a sensitive value, we should let the client provide the ARN of the secret, and use the data source as shown in the print.

3. This option is more complex because we need to understand where and how these values will come from, and adapt this for the `lambda_function` resource, which might take more time.

**Comment by zach-withcoherence** [comment]
OK, so here's what we want to do here
- add a new setting to the service here https://github.com/coherenceplatform/cnc/blob/main/src/cnc/models/config/settings.py#L217 that says "secrets_mode" which can be "arn" or "plaintext"
- in "arn" mode the value for each env var that is a secret will be the secret manager arn (we call item.secret_id) and the user will fetch the value in their code
- in "plaintext" mode we will call item.value on the secret and will pass the value as plaintext

let me know any questions. thanks!

**Comment by tmaior** [comment]
Comit: https://github.com/coherenceplatform/cnc/pull/127/commits/41bded3a33f015636492256cbfe4a387f4b98423

**Comment by zach-withcoherence** [comment]
@tmaior this is still not right. the place we need to update these is in the deploy script, not the terraform. there should be no terraform changes in this PR, only the deploy script changes!

File: src/cnc/flavors/aws/lambda-lite/1/deploy/env_vars.json.j2 [hunk 1]
```diff
@@ -1,7 +1,15 @@
 {
     "Variables": {
         {%- for item in service.insecure_environment_items %}
-        "{{ item.name }}": {{ item.value | tojson }}{%- if not loop.last %},{%- endif %}
+        "{{ item.name }}": {{ item.value | tojson }}{%- if not loop.last or service.environment_secrets %},{%- endif %}
+        {%- endfor %}
+
+        {%- for secret in service.environment_secrets %}
+        {% if service.settings.secrets_mode == 'plaintxt' %}
```
Comment (adamaziz15) [inline src/cnc/flavors/aws/lambda-lite/1/deploy/env_vars.json.j2:8]: This should be `plaintext` just missing an "e" in there ^

**Review by adamaziz15 — CHANGES_REQUESTED** [review]

**Review by adamaziz15 — APPROVED** [review]

**Review by zach-withcoherence — APPROVED** [review]

**Pull request merged**